runtimes/huggingface/tests/test_settings.py

```diff
@@ -0,0 +1,145 @@
+import pytest
+
+from typing import Dict, Any
+
+from mlserver.settings import ModelSettings, ModelParameters
+
+from mlserver_huggingface.runtime import HuggingFaceRuntime
+from mlserver_huggingface.settings import (
+    HuggingFaceSettings,
+    PARAMETERS_ENV_NAME,
+    get_huggingface_settings,
+    merge_huggingface_settings_extra,
+)
+from mlserver_huggingface.errors import MissingHuggingFaceSettings
+
+
+@pytest.fixture()
+def model_settings_extra_task():
+    return ModelSettings(
+        name="foo",
+        implementation=HuggingFaceRuntime,
+        parameters=ModelParameters(
+            extra={"task": "text-generation", "pretrained_model": "distilgpt2"}
+        ),
+    )
+
+
+@pytest.fixture()
+def model_settings_extra_none():
+    return ModelSettings(
+        name="foo",
+        implementation=HuggingFaceRuntime,
+        parameters=ModelParameters(extra=None),
+    )
+
+
+@pytest.fixture()
+def model_settings_extra_empty():
+    return ModelSettings(
+        name="foo",
+        implementation=HuggingFaceRuntime,
+        parameters=ModelParameters(extra={}),
+    )
+
+
+@pytest.mark.parametrize(
+    "model_settings,env_params,expected",
+    [
+        (
+            "model_settings_extra_task",
+            {"task": "question-answering"},
+            {"task": "question-answering", "pretrained_model": "distilgpt2"},
+        ),
+        (
+            "model_settings_extra_none",
+            {"task": "question-answering"},
+            {"task": "question-answering"},
+        ),
+        (
+            "model_settings_extra_empty",
+            {"task": "question-answering"},
+            {"task": "question-answering"},
+        ),
+    ],
+)
+def test_merge_huggingface_settings_extra(
+    model_settings: ModelSettings, env_params: Dict, expected: Dict, request: Any
+):
+    assert expected == merge_huggingface_settings_extra(
+        request.getfixturevalue(model_settings), env_params
```

Comment:
Interesting! I didn't know `request.getfixturevalue` was a thing!

Reply:
I was having issues with `lazy_fixture` so I went with this alternative! 😅